nsisplugin/IsServerCore.c

```diff
@@ -0,0 +1,72 @@
+#include <windows.h>
+#include <nsis/pluginapi.h>
+#include "Registry.h"
+#include "VersionInfo.h"
+
+typedef BOOL (WINAPI *_GetProductInfo)(DWORD, DWORD, DWORD, DWORD, PDWORD);
+
+static BOOL productInfoLoaded = FALSE;
+static _GetProductInfo $GetProductInfo;
+
+BOOL GetVistaProductInfo(DWORD dwOSMajorVersion, DWORD dwOSMinorVersion, DWORD dwSpMajorVersion, DWORD dwSpMinorVersion, PDWORD pdwReturnedProductType) {
+	if (!productInfoLoaded) {
+		productInfoLoaded = TRUE;
+		$GetProductInfo = (_GetProductInfo)GetProcAddress(GetModuleHandle(L"kernel32.dll"), "GetProductInfo");
+	}
+
+	if ($GetProductInfo) {
+		return $GetProductInfo(dwOSMajorVersion, dwOSMinorVersion, dwSpMajorVersion, dwSpMinorVersion, pdwReturnedProductType);
+	}
+
+	*pdwReturnedProductType = PRODUCT_UNDEFINED;
+	return FALSE;
+}
+
+PLUGIN_METHOD(IsServerCore) {
+	PLUGIN_INIT();
+
+    // Server Core introduced with Server 2008, skip on earlier versions.
+    if (!AtLeastWinVista()) {
+		pushint(0);
+		return;
+	}
+
+    // Server 2008 and 2008 R2 use GetProductInfo, 2012 and newer use the registry.
+    if (IsWinVista() || IsWin7()) {
+        DWORD productType = 0;
+        if (!GetVistaProductInfo(6, 0, 0, 0, &productType)) {
```

Comment:
Does this need to properly pass the version flags in, or does it not matter as long as the major digit is 6? (No idea why it even needs them…)